e2e/transactions/transactions_test.go

```diff
@@ -314,6 +314,89 @@ func (s *TransactionsTestSuite) TestSendEtherTx() {
+
+	s.StartTestBackend(params.StatusChainNetworkID)
+	defer s.StopTestBackend()
+
+	backend := s.LightEthereumService().StatusBackend
+	s.NotNil(backend)
+
+	// create an account
+	sampleAddress, _, _, err := s.Backend.AccountManager().CreateAccount(TestConfig.Account1.Password)
+	s.NoError(err)
+
+	completeQueuedTransaction := make(chan struct{})
+
+	// replace transaction notification handler
+	var txHash = gethcommon.Hash{}
+	signal.SetDefaultNodeNotificationHandler(func(jsonEvent string) { // nolint: dupl
+		var envelope signal.Envelope
+		err := json.Unmarshal([]byte(jsonEvent), &envelope)
+		s.NoError(err, fmt.Sprintf("cannot unmarshal JSON: %s", jsonEvent))
+
+		if envelope.Type == txqueue.EventTransactionQueued {
+			event := envelope.Event.(map[string]interface{})
+			log.Info("transaction queued (will be completed shortly)", "id", event["id"].(string))
+
+			// the first call will fail (we are not logged in, but trying to complete tx)
+			log.Info("trying to complete with no user logged in")
+			txHash, err = s.Backend.CompleteTransaction(
+				common.QueuedTxID(event["id"].(string)),
+				TestConfig.Account1.Password,
+			)
+			s.EqualError(
+				err,
+				account.ErrNoAccountSelected.Error(),
+				fmt.Sprintf("expected error on queued transaction[%v] not thrown", event["id"]),
+			)
+
+			// the second call will also fail (we are logged in as different user)
+			log.Info("trying to complete with invalid user")
+			err = s.Backend.AccountManager().SelectAccount(sampleAddress, TestConfig.Account1.Password)
+			s.NoError(err)
+			txHash, err = s.Backend.CompleteTransaction(
+				common.QueuedTxID(event["id"].(string)), TestConfig.Account1.Password)
+			s.EqualError(
+				err,
+				txqueue.ErrInvalidCompleteTxSender.Error(),
+				fmt.Sprintf("expected error on queued transaction[%v] not thrown", event["id"]),
+			)
+
+			// the third call will work as expected (as we are logged in with correct credentials)
+			log.Info("trying to complete with correct user, this should succeed")
+			s.NoError(s.Backend.AccountManager().SelectAccount(TestConfig.Account1.Address, TestConfig.Account1.Password))
+			txHash, err = s.Backend.CompleteTransaction(
+				common.QueuedTxID(event["id"].(string)),
+				TestConfig.Account1.Password,
+			)
+			s.NoError(err, fmt.Sprintf("cannot complete queued transaction[%v]", event["id"]))
+
+			log.Info("contract transaction complete", "URL", "https://ropsten.etherscan.io/tx/"+txHash.Hex())
+			close(completeQueuedTransaction)
+			return
+		}
+	})
+
+	// this call blocks, up until Complete Transaction is called
+	txHashCheck, err := s.Backend.SendTransaction(nil, common.SendTxArgs{
+		From:  common.FromAddress(TestConfig.Account1.Address),
+		To:    common.ToAddress(TestConfig.Account2.Address),
+		Value: (*hexutil.Big)(big.NewInt(1000000000000)),
+	})
+	s.NoError(err, "cannot send transaction")
+
+	select {
+	case <-completeQueuedTransaction:
+	case <-time.After(2 * time.Minute):
```

Comment:
As we use a private blockchain and it's supposed to be fast, maybe we can decrease it to seconds?

Reply:
Totally agree with you, more than that, If it were up to me, I would not use such a construction at all. Timeouts should be managed by context. Context is used everywhere across our calls, but for some reasons is not passed in the tests. The reason I didn't touch it is that It was said just to copy paste the test and make it pass on a private network as a basic showcase.

I would also split the test into 3-4 different test cases (with active user, with inactive user, with no user. It seems to me that there are too many asserts per test case right now

Reply:
I agree with all your points. We will start fixing our e2e tests when we have all pieces. This private network is one of them.